### PR TITLE
Publish incremental scan event during planning

### LINKED ISSUE #1711: Publish an incremental scan event during planning
There is a todo in the `IncrementalDataTableScan` to publish a incremental scan event. 

I can take a look into this. 
Incremental scan has a ‘from’ and ‘to’ snapshot that needs to be passed to the event as well. 

- We can add a new `IncrementalScanEvent` that has the ‘from’ and ‘to’ snapshot information as well in addition to the projection, filter and tablename info.  _Or_
- We could enhance the `ScanEvent` that exists today, although the snapshotId there won’t apply to the Incremental scan event.  What is the compatibility guarantees for ScanEvent across releases since it is part of the api folder?

@rdsr, @rdblue, please let me know your comments.  Thanks.

### PR DESCRIPTION
**Pull request by skambha**
Fixes #1711 

Changes include: 
- Create a new `IncrementalScanEvent` to keep track of the `from` and `to` snapshot id
- Publish the `IncrementalScanEvent` in` IncrementalDataTableScan` and remove the todo

Tests:
- Modify an existing test to check that the event is being published.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
You look to miss removing the TODO. Looks OK otherwise.

File: core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java [hunk 1]
```diff
@@ -78,8 +81,28 @@ public void testAppends() {
...
+
+      public IncrementalScanEvent event() {
+        return lastEvent;
+      }
+    }
+
+    MyListener listener1 = new MyListener();
+    Listeners.register(listener1, IncrementalScanEvent.class);
     filesMatch(Lists.newArrayList("B", "C", "D", "E"), appendsBetweenScan(1, 5));
+    Assert.assertTrue(listener1.event().fromSnapshotId() == 1);
```
Comment (HeartSaVioR) [inline core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java:101]: I'm in favor of check all fields as it's the only test verifying IncrementalScanEvent, but others may have different thought on this.

Comment (skambha) [reply core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java:101]: Thanks @HeartSaVioR!  I have updated the test to fetch and check all the fields of this event.

**Review by HeartSaVioR — APPROVED** [review]
LGTM

**Comment by skambha** [comment]
cc @rdblue

**Review by rdblue — APPROVED** [review]
Looks great! Thanks @skambha!

And thanks to @HeartSaVioR for reviewing. Good catch with the TODO.

**Pull request merged**